pkg/pgmodel/ingestor/series_writer.go

```diff
@@ -68,9 +63,10 @@ func (h *seriesWriter) WriteSeries(ctx context.Context, sv SeriesVisitor) error
 			info, ok := infos[metricName]
```

Comment:
`WriteSeries` is getting complex to understand. I feel we should break down this function at some point.